test/conformance/helpers/tracing/zipkin.go

```diff
@@ -47,9 +50,14 @@ var setTracingConfigOnce = sync.Once{}
 // TODO Do we need a tear down method to revert the config map to its original state?
 func setTracingConfigToZipkin(t *testing.T, client *lib.Client) {
 	setTracingConfigOnce.Do(func() {
+		// Get the namespace from the env key directly as system.Namespace() panics if unset
+		ns := os.Getenv(system.NamespaceEnvKey)
```

Comment:
Typically the system namespace is the namespace containing Knative Eventing controller and webhook deployments. This change allows zipkin to run in two possible namespaces:

- The system namespace alongside other Knative Eventing system components (i.e. `knative-eventing`)
- `istio-system`

Is the system namespace variable (`SYSTEM_NAMESPACE` I think) always set when this code runs? Is it even possible to use `istio-system`?

Reply:
sorry for the delayed response, I'm not too sure how to proceed with this to enable the desired behaviour without breaking the current functionality. iiuc, @n3wscott asks in this issue to get this value optionally from the `SYSTEM_NAMESPACE`, but then as you point out then it would always be set from there. Is that the expected behaviour @n3wscott? If not one possible alternative would be to use a different env variable? I'm open to other suggestions.

Reply:
in this case the system namespace is not what we want, that is the controlplane namespace and this component will be deployed to some other namespace potentially. Still reading...

Reply:
This is how serving does it: https://github.com/knative/serving/blob/47a96cc74e40f1990687aa00b7aacf1f9fac626b/vendor/knative.dev/pkg/tracing/config/tracing.go#L96-L100

Reply:
the config should come from the tracing config map. It might be a good idea to promote the code that handles that to PKG so serving and eventing can share the logic (and config maps if they are in the same namespace)